.github/workflows/deploy.yml

```diff
@@ -84,3 +84,29 @@ jobs:
         retry_on: error
         timeout_minutes: 120
         command: docker buildx build -f Dockerfile.ubuntu-xenial -t ${ORG}/github-runner:ubuntu-xenial --output "type=image,push=true" --platform linux/amd64,linux/arm64 .
+
+  debian_buster_deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Copy Repo Files
+      uses: actions/checkout@master
+    - name: Get GitHub organization or user
+      run: echo 'ORG='$(dirname ${GITHUB_REPOSITORY}) >> $GITHUB_ENV
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: crazy-max/ghaction-docker-buildx@v1
+      with:
+        buildx-version: latest
+    - name: Available platforms
+      run: echo ${{ steps.buildx.outputs.platforms }}
+    - name: Copy Dockerfile
+      run: cp Dockerfile Dockerfile.debian-buster; sed -i.bak "s/FROM.*/FROM ${ORG}\/github-runner-base:debian-buster/" Dockerfile.debian-buster
+    - name: Login
+      run: docker login -u ${{ secrets.DOCKER_USER }} -p ${{ secrets.DOCKER_TOKEN }}
+    - name: Build
+      uses: nick-invision/retry@v2
+      with:
+        max_attempts: 3
+        retry_on: error
+        timeout_minutes: 120
+        command: docker buildx build -f Dockerfile.debian-buster -t ${ORG}/github-runner:debian-buster --output "type=image,push=true" --platform linux/amd64,linux/arm64 .
```

Comment:
This doesn't seem to work:

```
#9 10.66 --------OS Information--------
#9 10.66 PRETTY_NAME="Debian GNU/Linux 10 (buster)"
#9 10.66 NAME="Debian GNU/Linux"
#9 10.66 VERSION_ID="10"
#9 10.66 VERSION="10 (buster)"
#9 10.66 VERSION_CODENAME=buster
#9 10.66 ID=debian
#9 10.66 HOME_URL="https://www.debian.org/"
#9 10.66 SUPPORT_URL="https://www.debian.org/support"
#9 10.66 BUG_REPORT_URL="https://bugs.debian.org/"
#9 10.66 ------------------------------
#9 10.66 The current OS is Debian based
#9 10.66 --------Debian Version--------
#9 10.67 10.10
#9 10.67 ------------------------------
#9 10.67 /usr/bin/apt-get
#9 10.84 Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
#9 10.84 Get:2 http://deb.debian.org/debian buster InRelease [122 kB]
#9 10.94 Get:3 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
#9 11.00 Ign:4 https://download.docker.com/linux/ubuntu buster InRelease
#9 11.03 Get:5 http://security.debian.org/debian-security buster/updates/main amd64 Packages [293 kB]
#9 11.06 Err:6 https://download.docker.com/linux/ubuntu buster Release
#9 11.06   404  Not Found [IP: 99.86.230.37 443]
#9 11.16 Get:7 http://deb.debian.org/debian buster/main amd64 Packages [7907 kB]
#9 12.27 Get:8 http://deb.debian.org/debian buster-updates/main amd64 Packages [15.2 kB]
#9 13.37 Reading package lists...
#9 14.07 E: The repository 'https://download.docker.com/linux/ubuntu buster Release' does not have a Release file.
#9 14.07 'apt-get' failed with exit code '0'
#9 14.07 Can't install dotnet core dependencies.
#9 14.07 You can manually install all required dependencies based on following documentation
#9 14.07 https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x
------
executor failed running [/bin/bash -o pipefail -c chmod +x /actions-runner/install_actions.sh   && /actions-runner/install_actions.sh ${GH_RUNNER_VERSION} ${TARGETPLATFORM}   && rm /actions-runner/install_actions.sh]: exit code: 1
```

To replicate:

```
± cp Dockerfile.base Dockerfile.base.debian-buster; sed -i.bak 's/FROM.*/FROM debian:buster/' Dockerfile.base.debian-buster
± docker build -f Dockerfile.base.debian-buster -t foo .
± cp Dockerfile Dockerfile.debian-buster; sed -i.bak "s/FROM.*/FROM foo/" Dockerfile.debian-buster
± docker build -f Dockerfile.debian-buster -t wut .
```

Reply:
My guess is from this line in the Dockerfile.base:

```
   && [[ $(lsb_release -cs) == "focal" ]] && ( add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu focal stable" ) ||     ( add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" ) \
```

if that gets too complicated it might be worthwhile to create a new Dockerfile.base specifically for debian as this one is pretty biased to ubuntu

Reply:
So I can fairly easily get the builds to finish, but they throw errors:
Either this:
```
./check_bindir "z$bindir" "z$execdir" "$bindir/git-add"
E: gnupg, gnupg2 and gnupg1 do not seem to be installed, but one of them is required for this operation
Removing intermediate container f57fe4020d27
 ---> f689a78d6a65
Successfully built f689a78d6a65
Successfully tagged foo:latest
$
```

Or if I change the code a little this:
```
./check_bindir "z$bindir" "z$execdir" "$bindir/git-add"
Warning: apt-key output should not be parsed (stdout is not a terminal)
Executing: /tmp/apt-key-gpghome.M7HEXCZ0o0/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 7EA0A9C3F273FCD8
gpg: key 8D81803C0EBFCD88: public key "Docker Release (CE deb) <docker@docker.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
Warning: apt-key output should not be parsed (stdout is not a terminal)
curl: (22) The requested URL returned error: 404 
gpg: no valid OpenPGP data found.
Removing intermediate container 132e4ea04a98
 ---> d1fa6477760d
Successfully built d1fa6477760d
Successfully tagged foo:latest
$ 
```

The ubuntu keyserver doesn't have the key `7EA0A9C3F273FCD8`.
Do you know what this step is even trying to do?

Reply:
thats the GPG public key for docker: https://askubuntu.com/questions/1347880/cant-install-docker-on-my-ubuntu-because-of-a-no-pubkey-7ea0a9c3f273fcd8-err

Reply:
But haven't we already installed docker without the key at this point?

My understanding is that we have a Docker container that start the runner and a Docker container inside that Docker container that is the actual runner. Are we failing to install Docker in the first Docker container then? Or are we trying to install Docker in the second Docker container even though it is not needed?

Reply:
Im not sure where you see that error, but:

It installs the GPG key [here](https://github.com/myoung34/docker-github-actions-runner/blob/master/Dockerfile.base#L53)
Then installs docker [here](https://github.com/myoung34/docker-github-actions-runner/blob/master/Dockerfile.base#L57)